sky/unit/test/rendering/flex_test.dart

```diff
@@ -47,10 +47,9 @@ void main() {
     RenderDecoratedBox box1 = new RenderDecoratedBox(decoration: new BoxDecoration());
     RenderDecoratedBox box2 = new RenderDecoratedBox(decoration: new BoxDecoration());
     RenderFlex flex = new RenderFlex();
-    flex.setupParentData(box2);
```

Comment:
I don't think we should break this. Being able to set up the parent data before you insert it was an intentional design decision.

I think the way to fix this is to have dropChild() set parentData to null.